dds/DCPS/Serializer.h

```diff
@@ -688,14 +688,15 @@ class OpenDDS_Dcps_Export Serializer {
   };
 
   template <typename T>
-  bool peek_helper(ACE_Message_Block* const block, T& t)
+  bool peek_helper(ACE_Message_Block* const block, size_t bytes, T& t)
   {
     bool result = false;
     char* const rd_ptr = block->rd_ptr();
-    if (block->cont()) {
-      result = peek_helper(block->cont(), t);
-    } else {
+    const size_t length = block->length();
+    if (!block->cont() || (block != 0 && bytes <= length)) {
```

Comment:
The `block != 0` expression doesn't look right because `block` was just dereferenced.